test/pkg.jl

```diff
@@ -139,6 +139,11 @@ temp_pkg_dir() do
     @test endswith(str, string(Pkg.installed("Example")))
     @test isempty(Pkg.dependents("Example"))
 
+    Pkg.checkout("Example", "test-branch") #17364
```

Comment:
this is more fragile than I'd like, there's no way to make a branch read only on github. does the remote have to be on github, or could it be a local filesystem remote?